src/Type/Php/ArrayMapFunctionReturnTypeExtension.php

```diff
@@ -37,13 +42,27 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 		$singleArrayArgument = !isset($functionCall->getArgs()[2]);
 		$callableType = $scope->getType($functionCall->getArgs()[0]->value);
 		$callableIsNull = $callableType->isNull()->yes();
+		$callback = null;
 
 		if ($callableType->isCallable()->yes()) {
 			$valueTypes = [new NeverType()];
 			foreach ($callableType->getCallableParametersAcceptors($scope) as $parametersAcceptor) {
 				$valueTypes[] = $parametersAcceptor->getReturnType();
 			}
 			$valueType = TypeCombinator::union(...$valueTypes);
+			$callback = $functionCall->getArgs()[0]->value;
+			if ($callback instanceof String_) {
```

Comment:
This entire code block isn't the right solution. You don't need to do any of that. You should be able to achieve the same thing with just:

```php
$callback = $functionCall->getArgs()[0]->value;
$callbackReturnType = $scope->getType(new FuncCall($callback, [
    new Arg(new TypeExpr($constantArray->getValueTypes()[$i])),
]));
```

If that doesn't work for some expressions, then the root cause is elsewhere.